src/graph.cpp

```diff
@@ -132,10 +127,10 @@ void Graph::visitedDFS(const std::shared_ptr<Vertex> &source) {
  * Time Complexity: O(|E|log|E|)
  */
 void Graph::kruskal(){
-    std::list<Edge *> edges; //TODO: better generate this list (edgeSet like vertexSet?)
-    for (Vertex *v : vertexSet){
-        //v->setVisited(false); //TODO: Check if this is necessary
-        for (const std::shared_ptr<Edge>& e: v->getAdj()) {
+    std::list<std::shared_ptr<Edge>> edges; //TODO: better generate this list (edgeSet like vertexSet?)
```

Comment:
We need a function here to add the edges that do not exist in the graph, with their distance, and possibly remove them later. Let me know what you think the best way to do this is

Reply:
### Taken care in call